cicecore/cicedynB/infrastructure/ice_grid.F90

```diff
@@ -406,6 +414,29 @@ subroutine init_grid2
          call rectgrid          ! regular rectangular grid
       endif
 
+      !-----------------------------------------------------------------
+      ! Diagnose OpenMP thread schedule, force order in output
+      !-----------------------------------------------------------------
+
+#if defined (_OPENMP)
```

Comment:
why is this done in ice_grid.F90?

Reply:
It's a good question.  It has to be done after the decomposition is established.  There is no obvious place for something like this in the code, but I'm happy to move it elsewhere.  I wanted to keep it out of the driver code because then it would need to be duplicated in all of them.  And it needed to be somewhere that is always called, so it couldn't be a "first call" thing in the column physics or dynamics.  I am open to moving this to another part of the code.  I think it's handy to have this information diagnosed, there is no way to tell otherwise.  In fact, I added this stuff because I wanted to verify the OMP schedule was working.  That's how I learned it wasn't and how we need to add the SCHEDULE(runtype) argument to the OMP pragma.  This is very confusing.  I pointed the problems out to the cheyenne help desk since they were helping me with some of this.  I proposed a change to the OpenMP standards to fix this and preserve backwards compatibility, but don't know if it'll go anywhere.

Reply:
I was thinking that it belonged in the module where decompositions are done (ice_blocks?).  But I guess that necessitates calling it from every driver.

Interesting that you proposed a change to the OpenMP standards!  We should track that in Consortium reporting docs.

Reply:
Do you have a link for your proposition @apcraig ? I'd like to follow any developments:)

Reply:
I forwarded my input to NCAR who has a person on the OpenMP committee.  Nothing more formal than that.